pkg/commatrix-creator/commatrix_test.go

```diff
@@ -0,0 +1,342 @@
+			gotComDetails, err := cm.getStaticEntries()
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			g.By("Comparing gotten static entries to wanted comDetails")
+			wantedComDetails := slices.Concat(types.BaremetalStaticEntriesMaster, types.GeneralStaticEntriesMaster)
+			o.Expect(gotComDetails).To(o.Equal(wantedComDetails))
+		})
+
+		g.It("Should successfully get static entries suitable to cloud standard cluster", func() {
+			g.By("Creating new communication matrix suitable to cloud standard cluster")
+			cm, err := New(nil, "", "", types.Cloud, types.Standard)
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			g.By("Getting static entries comDetails of the created communication matrix")
+			gotComDetails, err := cm.getStaticEntries()
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			g.By("Comparing gotten static entries to wanted comDetails")
+			wantedComDetails := slices.Concat(types.CloudStaticEntriesMaster, types.CloudStaticEntriesWorker,
+				types.GeneralStaticEntriesMaster, types.StandardStaticEntries, types.GeneralStaticEntriesWorker)
+			o.Expect(gotComDetails).To(o.Equal(wantedComDetails))
+		})
+
+		g.It("Should successfully get static entries suitable to cloud SNO cluster", func() {
+			g.By("Creating new communication matrix suitable to cloud SNO cluster")
+			cm, err := New(nil, "", "", types.Cloud, types.SNO)
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			g.By("Getting static entries comDetails of the created communication matrix")
+			gotComDetails, err := cm.getStaticEntries()
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			g.By("Comparing gotten static entries to wanted comDetails")
+			wantedComDetails := slices.Concat(types.CloudStaticEntriesMaster, types.GeneralStaticEntriesMaster)
+			o.Expect(gotComDetails).To(o.Equal(wantedComDetails))
+		})
+
+		g.It("Should return an error due to an invalid value for cluster environment", func() {
+			g.By("Creating new communication matrix with an invalid value for cluster environment")
+			cm, err := New(nil, "", "", -1, types.SNO)
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			g.By("Getting static entries comDetails of the created communication matrix")
+			gotComDetails, err := cm.getStaticEntries()
+			o.Expect(err).To(o.HaveOccurred())
+
+			g.By("Comparing gotten static entries to empty comDetails")
+			o.Expect(gotComDetails).To(o.Equal(nilComDetailsList))
+		})
+
+	})
+
+	g.Context("Create Endpoint Matrix", func() {
+		g.BeforeEach(func() {
+			sch := runtime.NewScheme()
+
+			err := corev1.AddToScheme(sch)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = discoveryv1.AddToScheme(sch)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			fakeClient := fake.NewClientBuilder().WithScheme(sch).WithObjects(testNode, testPod, testService, testEndpointSlice).Build()
+			fakeClientset := fakek.NewSimpleClientset()
+
+			clientset := &client.ClientSet{
+				Client:          fakeClient,
+				CoreV1Interface: fakeClientset.CoreV1(),
+			}
+
+			endpointSlices, err = endpointslices.New(clientset)
```

Comment:
These are empty right? There are no fake endpointslices in the test

Reply:
The fake client is initialized with "testEndpointslice" (see line 286)

Reply:
You are right sorry I missed that